File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        hash = [] # 원소값만 확인해도 되므로 List로 선언

        for i in nums:
            # 원소값이 hash에 있으면 True 리턴
            if i in hash:
                return True
            # 없으면 hash에 저장
            else:
                hash.append(i)
        return False

File: Data Structures & Algorithms/majority-element/submission-0.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> freqMap;  // <숫자, 등장 횟수>

        /* 배열을 순회하면서 해시맵 채우기 */
        for(int i=0; i<size(nums); i++)
           freqMap[nums[i]]++;

        int result = 0;
        int maxCount = 0;
        for(auto const& [val, count] : freqMap) {
            if(count > maxCount) {
                maxCount = count;
                result = val;
            }
        }
        return result;
    }
};